### ISSUE TITLE
[User] : Missing email in User Information keeps User tab from loading

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#21308]
Found when working on Support ticket: [105801](https://dotcms.zendesk.com/agent/tickets/105801)
Reproducible as of dotCMS 5.3.8.6.1, up to latest master (21.11)

For unknown reasons -- specially with customers coming from very old versions of dotCMS -- there are cases in which Users don't have any email address. This is an issue when the User missing such a piece of data is the `system user`, which cannot be edited via the UI in the back-end.

**To Reproduce**
1. Set up a dotCMS 5.3.8.6.1 instance with the Demo Starter.
2. Let it initialize properly, and shut it down.
3. Via SQL queries, update for the `system` User to have an empty email address:
```sql
UPDATE user_ SET emailaddress = NULL WHERE userid = 'system';
```
4. Start up dotCMS again, and go to the **Roles & Tools** portlet.
5. Look for the `CMS Administrator` Role, and lick it.
6. You'll get the following Java exception in the log:
```
[24/11/21 18:07:37:449 GMT] INFO log.accessLog: Method execution failed:
java.lang.NullPointerException: null
at com.liferay.portal.model.User.toMap(User.java:395) ~[dotcms_5.3.8.6.1_999999.jar:?]
at com.dotmarketing.business.ajax.RoleAjax.getUsersByRole(RoleAjax.java:197) ~[dotcms_5.3.8.6.1_999999.jar:?]
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_282]
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_282]
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_282]
at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_282]
at com.dotcms.repackage.org.directwebremoting.impl.CreatorModule$1.doFilter(CreatorModule.java:229) ~[dot.dwr-3rc2modified_3.jar:?]
...
...
```

**Expected behavior**
The application should fail gracefully, and indicate the respective error message in the UI, the `dotcms.log` file, or both.

**Additional context**
The currently valid workaround is to run the SQL query to set an email address for the failing User:
```sql
UPDATE user_ SET emailaddress = 'tempaddress@fakeemail.com' WHERE userid = '{USERID-TO-UPDATE}';
```

**Comment by alfredo-dotcms** [comment]
![image](https://user-images.githubusercontent.com/37185433/145308737-6086ad88-b79c-4883-88bd-5dabcc0033c3.png)

**Comment by alfredo-dotcms** [comment]
looks good to me!

**Comment by bryanboza** [comment]
Fixed, tested on release-22.02 // Docker // FF. I'm getting the same message of the above image
